src/software/ai/navigator/obstacle/obstacle_generation_test.cpp

```diff
@@ -0,0 +1,19 @@
+#include "software/ai/navigator/obstacle/obstacle_generation.h"
+
+#include <gtest/gtest.h>
+#include <math.h>
+
+#include <cmath>
+#include <cstdlib>
+#include <ctime>
+#include <sstream>
+
+#include "software/geom/point.h"
+#include "software/geom/util.h"
+
+int main(int argc, char **argv)
```

Comment:
This isn't built into a test library?

Reply:
I accidentally included this whoops. This was transferred from nav util test but I had to remove the calculatetransitionspeed test because I made it private to navigator

Reply:
Testing for obstacle generation should be done after obstacle rewrite after new geom imo

Reply:
I'm tempted to remove the file completely

Reply:
If this file isn't being used then yea it should be removed, but we should keep the tests for `calculateTransitionSpeed` so we don't have to rewrite them.

Reply:
How do we call them if they're private within navigator?

Reply:
Either make it protected and make the test fixture a friend class, or just make it public for now. I'd rather it be public and tested than private and untested.

Reply:
As a note, not sure if this is the case here, but if you find that there's private functions you really want to test, that's a good sign the class is doing too much and should be broken up.

Reply:
Arguably no, this doesn't need testing. It is a single line of code that could even be inlined.

Reply:
I think inlining it would make it more confusing, since I don't think it would be immediately obvious what a big line of projections and scaling are doing. I think it totally makes sense in it's own function, since it has the potential to be used by other things (maybe) and it makes it way more readable.

Reply:
oh no, I'm not proposing inlining it. I'm saying that since it's one line of code, it really doesn't need a set of unit tests

Reply:
> if you find that there's private functions you really want to test

Not actually necessary to test it since the functionality should be entirely covered by geom tests on project and norm

Reply:
There is _definitely_ enough complexity in that function that it should be thoroughly tested. I don't think it's sufficient to say that the geom tests cover this as well, we're combining 5-6 operators/functions! The fact that there's enough complexity and it's only one line could indicate that this should be broken up into multiple lines. Out-of-scope for this PR, and not a big deal, but just a note.